include/dlaf/matrix/matrix.h

```diff
@@ -333,6 +335,38 @@ Matrix<T, device> createMatrixFromTile(const GlobalElementSize& size, const Tile
   return createMatrixFromTile<device>(size, block_size, ld_tile, tiles_per_col, comm, {0, 0}, ptr);
 }
 
+namespace internal {
+template <class RetT, class T, dlaf::Device d, class RangeTile>
+auto select_generic(Matrix<T, d>& matrix, RangeTile range,
+                    RetT (Matrix<T, d>::*f)(const typename RangeTile::index2d_t&)) {
+  std::vector<RetT> tiles;
```

Comment:
With the assumption that the range has random access iterators:
```suggestion
  std::vector<RetT> tiles;
  tiles.reserve(std::distance(std::begin(range), std::end(range));
```
Unlikely to make a big difference, but doesn't hurt.

Reply:
In order to do it, I will have to check what are the requirements of `std::distance` for the iterator (they should be quite basic requirements), and it is somehow linked to what I wanted to do when i wrote this: standardize a bit our IterableRange2D iterator.

I'll check it soon 😉 

> note-to-self: in case we make `IterableRange2D` more "standard", i.e. compatible with std iterators traits (at least a subset of them), we could use any kind of iterable thing as range (e.g. we could create a vector with a custom set of indexes and pass that).

Reply:
As said in a meeting, it would not be so straightforward to adapt `IterableRange2D` to be an [`InputIterator`](https://en.cppreference.com/w/cpp/named_req/InputIterator) (as requested by `std::distance`) and at the same time it would not be so clean to compute the distance (e.g. we should convert the result of the range boundaries to a linear index).

For reference, see also https://github.com/eth-cscs/DLA-Future/pull/335#issuecomment-822397375

Reply:
Following what you said, the IteratorRange2D has been "standardized" a bit and now it is possible to do what you suggested here.